internal/rbd/driver/driver.go

```diff
@@ -213,7 +210,7 @@ func (r *Driver) setupCSIAddonsServer(conf *util.Config) error {
 	r.cas.RegisterService(is)
 
 	if conf.IsControllerServer {
-		rs := casrbd.NewReclaimSpaceControllerServer()
+		rs := casrbd.NewReclaimSpaceControllerServer(NewControllerServer(r.cd))
```

Comment:
```suggestion
		rs := casrbd.NewReclaimSpaceControllerServer(r.cs)
```

Reply:
Yes, we should use the same already created Controller and Node Server struct, as they will share the same sets of locks. Since the ReplicationServer uses a new ControllerServer struct, I decided to follow the same for ReclaimSpace ControllerServer/NodeServer.

Reply:
lets discuss on this one and we need to see can what will happen when we try to do nodeserver operations like (expand/umount) etc when the fstrim is going on can it leads to corruption etc. based on that we can have central/shared locking. Is it allowed to expand or remove the image when the ControllerReclaimOperation is going on etc.